Source/Core/Transforms.js

```diff
@@ -344,6 +346,68 @@ define([
         return result;
     };
 
+    var scratchHPRQuaternion = new Quaternion();
+    var scratchScale = new Cartesian3(1.0, 1.0, 1.0);
+    var scratchHPRMatrix4 = new Matrix4();
+
+    /**
+     * Computes a 4x4 transformation matrix from a reference frame with axes computed from the heading-pitch-roll angles
+     * centered at the provided origin to the provided ellipsoid's fixed reference frame.
+     *
+     * @param {Cartesian3} origin The center point of the local reference frame.
+     * @param {Number} heading The heading angle in radians.
```

Comment:
Here and throughout the doc in this PR, we need to describe what `heading`, `pitch`, and `roll` are, e.g., relative to north, geodetic normal, etc.  If you want to go all out, draw a picture.